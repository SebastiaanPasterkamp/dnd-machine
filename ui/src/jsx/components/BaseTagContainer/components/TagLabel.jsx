import React from 'react';
import PropTypes from 'prop-types';

export const TagLabel = function({ children }) {
    return (
        <span className="nice-tag-label">
            {children}
        </span>
    );
};

TagLabel.propTypes = {
    children: PropTypes.node,
};

export default TagLabel;
