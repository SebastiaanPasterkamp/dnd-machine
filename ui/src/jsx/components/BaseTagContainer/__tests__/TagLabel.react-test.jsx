import React from 'react';
import renderer from 'react-test-renderer';

import TagLabel from '../components/TagLabel';

describe('Component: TagLabel', () => {
    it('works with minimum props', () => {
        const tree = renderer.create(
            <TagLabel />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('works with all props', () => {
        const tree = renderer.create(
            <TagLabel>
                Hello Tag
            </TagLabel>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
