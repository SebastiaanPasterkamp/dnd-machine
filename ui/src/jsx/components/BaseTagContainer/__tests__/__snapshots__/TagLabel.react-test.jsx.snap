// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TagLabel works with all props 1`] = `
<span
  className="nice-tag-label"
>
  Hello Tag
</span>
`;

exports[`Component: TagLabel works with minimum props 1`] = `
<span
  className="nice-tag-label"
/>
`;
